.setup/INSTALL_SUBMITTY_HELPER_BIN.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+########################################################################################################################
+########################################################################################################################
+# COPY VARIOUS SCRIPTS USED BY INSTRUCTORS AND SYS ADMINS FOR COURSE ADMINISTRATION
+
+echo -e "Copy the user scripts"
+
+if [ -z ${SUBMITTY_INSTALL_DIR+x} ]; then
```

Comment:
is the "+x" a piece of magic bash syntax?

Reply:
Yeah. Copying my answer from above:

The `-z` tests for not set or empty. The variable is then read as `${VARIABLE+[word]}` which is that if `${VARIABLE}` is null/empty, replace it with nothing, else replace it with `[word]` (or in this case, `x`). The check itself is so that if the variable isn't set (because we're running this standalone), then we enter the block and fill in the variables as needed. We do need the `+[word]` because `-z` cannot distinguish between the empty and not set cases.

Basically, I hate the convoluted bash syntax for doing stuff that's super simple in other languages.

Reply:
ok, sorry i missed your answer.  argh to both bash & github.